xarray_beam/_src/core_test.py

```diff
@@ -342,6 +418,86 @@ def test_task_count(self):
     to_chunks = xbeam.DatasetToChunks(dataset, chunks={'z': 1}, split_vars=True)
     self.assertEqual(to_chunks._task_count(), 12)
 
+    datasets = [dataset.copy() for _ in range(9)]
+
+    to_chunks = xbeam.DatasetToChunks(datasets, chunks={'x': 1})
+    self.assertEqual(to_chunks._task_count(), 3)
+
+    to_chunks = xbeam.DatasetToChunks(
+        datasets, chunks={'x': 1}, split_vars=True
+    )
+    self.assertEqual(to_chunks._task_count(), 7)
+
+    to_chunks = xbeam.DatasetToChunks(
+        datasets, chunks={'y': 1}, split_vars=True
+    )
+    self.assertEqual(to_chunks._task_count(), 8)
+
+    to_chunks = xbeam.DatasetToChunks(
+        datasets, chunks={'z': 1}, split_vars=True
+    )
+    self.assertEqual(to_chunks._task_count(), 12)
+
+  def test_validate(self):
+    dataset = xarray.Dataset({
+        'foo': (('x', 'y'), np.zeros((3, 6))),
+        'bar': ('x', np.zeros(3)),
+        'baz': ('z', np.zeros(10)),
+    })
+
+    ## TODO(alxr): reproduce error
```

Comment:
@shoyer I could use your help with this test case. From my read, it looks like when chunks aren't set, they are no longer set as `None`. Instead, they become `Froze({})`. 

What should the correct assertion be in the constructor? i.e. see [line 303](https://github.com/google/xarray-beam/pull/69/files#diff-a752981324f7af231a6cf543a760ebbb8735d4fd76e3245c504777861714d167R303). I am uncertain about the intended behavior.

Reply:
This commented out test case looks correct to me.

The validation test is definitely broken, though, because indeed `Dataset.chunks` always returns a mapping. Let's switch it to use `if chunks` instead.

Reply:
This commented out test case looks correct to me.

The validation test is definitely broken, though, because indeed `Dataset.chunks` always returns a mapping. Let's switch it to use `if chunks` instead.

Reply:
This commented out test case looks correct to me.

The validation test is definitely broken, though, because indeed `Dataset.chunks` always returns a mapping. Let's switch it to use `if chunks` instead.

Reply:
This commented out test case looks correct to me.

The validation test is definitely broken, though, because indeed `Dataset.chunks` always returns a mapping. Let's switch it to use `if chunks` instead.

Reply:
This commented out test case looks correct to me.

The validation test is definitely broken, though, because indeed `Dataset.chunks` always returns a mapping. Let's switch it to use `if chunks` instead.